### ISSUE TITLE
[Accessibility] Text-based input placeholders (excluding `Search`) fail colour contrast validation

### ISSUE THREAD

**Issue by damienrobson-sage** [Sage/carbon#6587]
### Description

There is an accessibility failure regarding the colour contrast of placeholders in text boxes (with the exception of the `Search` component). Using a colour contrast checker, the value returned sits at 2.1, far short of the 4.5 required to be deemed accessible.

### Reproduction

https://stackblitz.com/edit/placeholderaccessibility?file=src%2FApp.tsx

### Steps to reproduce

Using Chrome's shadow DOM, you can inspect the colour of the placeholder text which looks to be set to `var(--colorsUtilityYin030)`. I ran the hexadecmial version of it through [this tool](https://www.joedolson.com/tools/color-contrast.php?type=hex&color=%23000000&color2=%23ffffff&alpha=0.3) to arrive at the 2.1:1 ratio above.

### Suggested solution

Given that placeholder text must meet a minimum colour contrast ratio of at least 4.5:1 ([WCAG 1.4.3](https://www.w3.org/TR/WCAG21/#contrast-minimum)), the same colour used for the hint text (`var(--colorsUtilityYin055)`) could be used (as this has a ratio of 4.74

### Carbon version

latest

### Relevant browsers

Chrome

### Relevant OSs

MacOS

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by edleeks87** [comment]
FE-6379
